### PR TITLE
vmray: record command line info

### PR DESCRIPTION
**Pull request by mr-tz**
example results afterwards:

image

image

### Checklist

- [ ] No CHANGELOG update needed

- [x] No new tests needed

- [ ] No documentation update needed

### CODE REVIEW

File: capa/features/extractors/vmray/__init__.py [hunk 1]
```diff
@@ -160,7 +170,12 @@ def _compute_monitor_processes(self):
                 self.sv2.processes[process.ref_parent_process.path[1]].os_pid if process.ref_parent_process else 0
             )
             self.monitor_processes[process.monitor_id] = VMRayMonitorProcess(
-                process.os_pid, ppid, process.monitor_id, process.image_name
+                process.os_pid,
+                ppid,
+                process.monitor_id,
+                process.image_name,
+                sanitize_string(process.filename),
+                sanitize_string(process.cmd_line),
```
Comment (mike-hunhoff) [inline capa/features/extractors/vmray/__init__.py:178]: Is there a reason not to handle this via the `SanitizedString` as is done with `MonitorProcess`?

Comment (mr-tz) [reply capa/features/extractors/vmray/__init__.py:178]: `VMRayMonitorProcess` is this file's local dataclass (with no validators)

Comment (mike-hunhoff) [reply capa/features/extractors/vmray/__init__.py:178]: Apologies, I should have been more specific. I meant add the `SanitizedString` validator to `Process`:
https://github.com/mandiant/capa/blob/7a8551bef997b1f309a052559b29e8e9556eea57/capa/features/extractors/vmray/models.py#L311-L321
Like you've done for `MonitorProcess`:
https://github.com/mandiant/capa/blob/7a8551bef997b1f309a052559b29e8e9556eea57/capa/features/extractors/vmray/models.py#L148-L164

Comment (mr-tz) [reply capa/features/extractors/vmray/__init__.py]: ah, of course, good catch, see updated PR

**Review by mike-hunhoff — APPROVED** [review]
🚀

File: capa/features/extractors/vmray/models.py [hunk 2]
```diff
@@ -136,19 +136,28 @@ class FunctionReturn(BaseModel):
     from_addr: HexInt = Field(alias="from")
 
 
+def sanitize_string(value: str) -> str:
+    # e.g. "cmd_line": "\"C:\\Users\\38lTTV5Kii\\Desktop\\filename.exe\" ",
+    return value.replace("\\\\", "\\").strip(' "')
```
Comment (williballenthin) [inline capa/features/extractors/vmray/models.py:141]: maybe json decode a portion of the string?

**Review by williballenthin — APPROVED** [review]

File: capa/features/extractors/vmray/models.py (hunk 2 above)
Comment (mr-tz) [reply capa/features/extractors/vmray/models.py:141]: here are more examples of items we could find here:
```
"cmd_line": "\"C:\\Users\\38lTTV5Kii\\Desktop\\filename.exe\" ",
"cmd_line": "\"C:\\Users\\8qy2SK\\AppData\\Local\\Temp\\Zip.exe\" ",
"cmd_line": "\"C:\\Users\\8qy2SK\\Desktop\\mulvpilibfy.exe\" ",
"cmd_line": "\"C:\\Users\\WhuOXYsD\\Desktop\\filename.exe\" ",
"cmd_line": "C:\\Windows\\system32\\cmd.exe /c tasklist",
"cmd_line": "C:\\Windows\\system32\\svchost.exe -k netsvcs -p",
"cmd_line": "C:\\Windows\\system32\\wbem\\wmiprvse.exe -secured -Embedding",
"cmd_line": "/c wmic diskdrive get model",
"cmd_line": "tasklist",
"cmd_line": "wmic  diskdrive get model",
```

can you clarify the json decode part?

Comment (williballenthin) [reply capa/features/extractors/vmray/models.py:141]:
```py
# like: "cmd_line": "\"C:\\Users\\38lTTV5Kii\\Desktop\\filename.exe\" ",

# like: "\"C:\\Users\\38lTTV5Kii\\Desktop\\filename.exe\" ",
cmd_line = line.partition(": ")[2]

# like: r'"C:\Users\38lTTV5Kii\Desktop\filename.exe" '
cmd_line = json.loads(cmd_line.rstrip(","))
```

alternatively:

```py
cmd_line = json.loads("{" + line + "}")["cmd_line"]
```

Comment (mr-tz) [reply capa/features/extractors/vmray/models.py:141]: I still don't follow. But my examples aren't fully clear. For the sanitization we only are dealing with the values.

image

```python
json.loads("\"C:\\Users\\38lTTV5Kii\\Desktop\\filename.exe\" ")
...
json.decoder.JSONDecodeError: Invalid \escape: line 1 column 4 (char 3)
```

Comment (williballenthin) [reply capa/features/extractors/vmray/models.py:141]: ok, cool, i misunderstood the data to be json-ish, but i see now.

**Pull request merged**